As per http://bugs.python.org/issue3588, LINKFORSHARED is (probably) not
intended to be part of a public API, and its value cannot be relied on other
than during a build of Python itself.

--- gdb/python/python-config.py.dist	2014-10-31 01:29:58.000000000 +0000
+++ gdb/python/python-config.py	2014-10-31 01:30:10.000000000 +0000
@@ -72,7 +72,5 @@ for opt in opt_flags:
                     libs.insert(0, '-L' + getvar('LIBPL'))
                 elif os.name == 'nt':
                     libs.insert(0, '-L' + sysconfig.PREFIX + '/libs')
-            if getvar('LINKFORSHARED') is not None:
-                libs.extend(getvar('LINKFORSHARED').split())
         print (to_unix_path(' '.join(libs)))
 
